src/IR/LowerAst.hs

```diff
@@ -162,6 +162,14 @@ lowerExpr (A.OpRegion _ _) =
 lowerExpr (A.Match s ps) =
   I.Match <$> lowerExpr s <*> mapM lowerArm ps <*> pure untyped
   where lowerArm (a, e) = (,) <$> lowerPatAlt a <*> lowerExpr e
+lowerExpr (A.Tuple es) =
+  apply_recurse (I.Data "Pair" untyped) <$> mapM lowerExpr es
+
+
+apply_recurse :: I.Expr I.Annotations -> [I.Expr I.Annotations] -> I.Expr I.Annotations
+apply_recurse e [] = e
+apply_recurse e (x:xs) = apply_recurse (I.App e x untyped) xs
```

Comment:
Is it possible to use zipApp from IR.hs instead of defining a new function?

Reply:
find it a bit hard to use that since needs some complex conversions of expression list.

Reply:
I move apply_recurse to a where clause so it can be used as a helper function.